File: src/main/java/de/mikeyllp/miniGamesV4/commands/admincommands/clearCommand.java
package de.mikeyllp.miniGamesV4.commands.admincommands;

import dev.jorel.commandapi.CommandAPICommand;
import org.bukkit.plugin.java.JavaPlugin;

import static de.mikeyllp.miniGamesV4.utils.MessageUtils.sendNoPermissionMessage;
import static de.mikeyllp.miniGamesV4.utils.clearUtils.clearAllLists;

public class clearCommand extends CommandAPICommand {
    public clearCommand(String commandName, JavaPlugin plugin) {
        super(commandName);
        executes(((sender, args) -> {
            if (!sender.hasPermission("minigamesv4.admin")) {
                sendNoPermissionMessage(sender);
                return;
            }
            clearAllLists(sender, plugin);
        }));
    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/commands/admincommands/clearCommand.kt
package de.mikeyllp.miniGamesV4.commands.admincommands

import de.mikeyllp.miniGamesV4.utils.MessageUtils
import de.mikeyllp.miniGamesV4.utils.MinigamesPermissionRegistry
import de.mikeyllp.miniGamesV4.utils.clearUtils
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.subcommand

fun CommandAPICommand.clearCommand() = subcommand("clear") {
    anyExecutor { sender, args ->
        if (!sender.hasPermission(MinigamesPermissionRegistry.COMMAND_CLEAR)) {
            MessageUtils.sendNoPermissionMessage(sender)
            return@anyExecutor
        }

        clearUtils.clearAllLists(sender)
    }
}